politeiad/cache/cache.go

```diff
@@ -0,0 +1,188 @@
+	MIME    string // MIME type
+	Digest  string // SHA256 of decoded Payload
+	Payload string // base64 encoded file
+}
+
+// MetadataStream identifies a metadata stream by its identity.
+type MetadataStream struct {
+	ID      uint64 // Stream identity
+	Payload string // String encoded metadata
+}
+
+// CensorshipRecord contains the proof that a record was accepted for review.
+// The proof is verifiable on the client side.  The Merkle field contains the
+// ordered merkle root of all files in the record. The Token field contains a
+// random censorship token that is signed by the server private key.  The token
+// can be used on the client to verify the authenticity of the
+// CensorshipRecord.
+type CensorshipRecord struct {
+	Token     string // Censorship token
+	Merkle    string // Merkle root of record
+	Signature string // Signature of merkle+token
+}
+
+// Record is an entire record and it's content.
+type Record struct {
+	Version          string           // Version of this record
+	Status           RecordStatusT    // Current status
+	Timestamp        int64            // Last update
+	CensorshipRecord CensorshipRecord // Censorship record
+	Metadata         []MetadataStream // Metadata streams
+	Files            []File           // Files that make up the record
+}
+
+// InventoryStats is a summary of the number of records in the cache grouped
+// by record status.  Only the latest version of each record is included.
+type InventoryStats struct {
+	Invalid           int // Number of invalid records
+	NotReviewed       int // Number of unreviewed records
+	Censored          int // Number of censored records
+	Public            int // Number of public records
+	UnreviewedChanges int // Number of unreviewed records with edits
+	Archived          int // Number of archived records
+}
+
+// PluginCommand is used to execute a plugin command.  The reply payload
+// contains the reply from politeiad, which is sometimes required by commands
+// that write data to the cache.  The reply payload will be empty for commands
+// that only read data from the cache.
+type PluginCommand struct {
+	ID             string // Plugin identifier
+	Command        string // Command identifier
+	CommandPayload string // Command payload
+	ReplyPayload   string // Command reply payload
+}
+
+// PluginCommandReply is used to reply to a PluginCommand.
+type PluginCommandReply struct {
+	ID      string // Plugin identifier
+	Command string // Command identifier
+	Payload string // Actual command reply
+}
+
+// PluginSetting is a structure that holds key/value pairs of a plugin setting.
+type PluginSetting struct {
+	Key   string // Name of setting
+	Value string // Value of setting
+}
+
+// Plugin describes a plugin and its settings.
+type Plugin struct {
+	ID       string          // Identifier
+	Version  string          // Version
+	Settings []PluginSetting // Settings
+}
+
+// PluginDriver is the interface that all cache plugins must implement.
```

Comment:
Explain what this does and why it exists.

Reply:
Added doco.